src/vcf/validate_optional_policy.cpp

```diff
@@ -146,6 +152,41 @@ namespace ebi
         }
     }
 
+    void ValidateOptionalPolicy::check_body_entry_alt_gvcf_gt_value(ParsingState & state, Record const & record) const
+    {
+        if (std::find(record.alternate_alleles.begin(), record.alternate_alleles.end(), GVCF)
+            != record.alternate_alleles.end() && record.format[0] == vcf::GT) {
+            for (auto & sample : record.samples) {
+                std::string gt_subfield = sample.substr(0, sample.find(':'));
+                std::vector<std::string> alleles;
+                util::string_split(gt_subfield, "/|", alleles);
+                bool found_in_sample = true;
+                for (auto & allele : alleles) {
+                    if (allele != "0") {
+                        found_in_sample = false;
+                        break;
+                    }
+                }
```

Comment:
what about extracting a function since the substr until here? so that this function is just
```
if (...) {
    for (sample ...) {
        if (sample_has_reference_in_all_alleles(sample)) {
            return;
        }
    }
    throw ...
}
``` 
which reads quite nice